File: Python/1310.py
from itertools import accumulate
from operator import xor
from typing import List

class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        arr[:] = accumulate(arr, xor)
        ans = []

        for q in queries:
            if q[0] == 0:
                ans.append( arr[q[1]] )
            else:
                ans.append( arr[q[1]]^arr[q[0]-1] )

        return ans
    

# Topic: Bit Manipulation
# Time Complexity: O(n + m)
# Space Complexity: O(1)     

File: C++/1310.cpp
#include <vector>


class Solution {
public:
    std::vector<int> xorQueries(std::vector<int>& arr, std::vector<std::vector<int>>& queries) {
        std::vector<int> prefix(arr.size());
        std::vector<int> ans(queries.size());
        prefix[0] = arr[0];

        for(int i = 1; i < arr.size(); ++i){
            prefix[i] = prefix[i-1] ^ arr[i];
        }

        for(int i = 0; i < queries.size(); ++i){
            int left = queries[i][0];
            int right = queries[i][1];

            if (left == 0){
                ans[i] = prefix[right];
            } else {
                ans[i] = prefix[right] ^ prefix[left-1];
            }
        }
        
        return ans;
    }
};

// Topic: Bit Manipulation
// Time Complexity: O(n + m)
// Space Complexity: O(n + m)